### PR TITLE
fixed https://github.com/NuGet/Home/issues/1039

### PR DESCRIPTION
**Pull request by zhili1208**
https://github.com/NuGet/Home/issues/1039
won't fix the error message should be associated with project in error List here.
added project name to error message and handle the exception

### CODE REVIEW

**Comment by zhili1208** [comment]
@emgarten @yishaigalatzer @feiling @deepakaravindr

File: src/NuGet.Commands/Strings.resx [hunk 1]
```diff
@@ -151,7 +151,7 @@
     <value>Some packages are not compatible with {0}.</value>
   </data>
   <data name="Log_ProjectDoesNotSpecifyTargetFrameworks" xml:space="preserve">
-    <value>The project does not specify any target frameworks.</value>
+    <value>The project {0} does not specify any target frameworks in project.json.</value>
```
Comment (emgarten) [inline src/NuGet.Commands/Strings.resx]: Add single quotes around the project name like this '{0}'   It makes it easier to understand the message when the project has an odd name, or a name that blends into the sentence.

File: src/NuGet.Commands/RestoreCommand.cs [hunk 2]
```diff
@@ -400,7 +400,7 @@ private bool ResolutionSucceeded(IEnumerable<RestoreTargetGraph> graphs)
         {
             // Get the project graph
             var projectFrameworks = project.TargetFrameworks.Select(f => f.FrameworkName).ToList();
-            if (projectFrameworks.Count > 1)
+            if (projectFrameworks.Count > 1 || targetGraphs.Count() == 0)
```
Comment (emgarten) [inline src/NuGet.Commands/RestoreCommand.cs]: You don't need to enumerate the whole list here, you can use: !targetGraphs.Any()

File: src/NuGet.Commands/RestoreCommand.cs [hunk 3]
```diff
@@ -149,7 +149,7 @@ public Task<RestoreResult> ExecuteAsync()
         {
             if (_request.Project.TargetFrameworks.Count == 0)
             {
-                _log.LogError(Strings.Log_ProjectDoesNotSpecifyTargetFrameworks);
+                _log.LogError(Strings.FormatLog_ProjectDoesNotSpecifyTargetFrameworks(_request.Project.Name));
```
Comment (emgarten) [inline src/NuGet.Commands/RestoreCommand.cs:152]: lowercase s on string as per the convention. 

What is doing the formatting here? is it using the current culture?

**Comment by yishaigalatzer** [comment]
String changes need pre-approval. I'm ok with this one

**Comment by zhili1208** [comment]
@emgarten Yes, it's using the current culture, 
string.Format(CultureInfo.CurrentCulture, GetString("Log_ProjectDoesNotSpecifyTargetFrameworks"), p0);

**Comment by emgarten** [comment]
:shipit:

**Pull request closed without merging**